### PR TITLE
Fix no-engine

### PR DESCRIPTION
**Pull request by mattcaswell**
Make sure references to ENGINE functions are appropriately guarded.

### CODE REVIEW

**Comment by mattcaswell** [comment]
As a side note to this PR, I find it strange that in some places we use `ENGINE_get_name()` to get ENGINE information to be passed to the providers, while in others we use `ENGINE_get_id()`.

**Comment by levitte** [comment]
> As a side note to this PR, I find it strange that in some places we use `ENGINE_get_name()` to get ENGINE information to be passed to the providers, while in others we use `ENGINE_get_id()`.

Agreed. I wonder why we have two functions with the same purpose...

File: crypto/evp/p_lib.c [hunk 1]
```diff
@@ -321,7 +321,9 @@ EVP_PKEY *EVP_PKEY_new_CMAC_key(ENGINE *e, const unsigned char *priv,
                                 size_t len, const EVP_CIPHER *cipher)
 {
 #ifndef OPENSSL_NO_CMAC
+# ifndef OPENSSL_NO_ENGINE
     const char *engine_name = e != NULL ? ENGINE_get_name(e) : NULL;
```
Comment (levitte) [inline crypto/evp/p_lib.c]: `ENGINE_get_name(e)` is wrong and should be `ENGINE_get_id(e)`

**Comment by levitte** [comment]
With engines, one might compare "id" and "name" with short and long name in the object db

**Comment by mattcaswell** [comment]
I pushed a new commit to convert the `ENGINE_get_name()` call to `ENGINE_get_id()`. Please take a look.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**